### PR TITLE
Add str_to_rcode function

### PR DESCRIPTION
**Pull request by hstern**

### CODE REVIEW

File: t/test-str_to_rcode.c [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+#include <ctype.h>
+
+#include <libmy/ubuf.h>
+#include <wdns.h>
+
+#define NAME "test-str_to_rcode"
+
+static struct u16str {
+    uint16_t u16;
+    const char *str;
```
Comment (edmonds) [inline t/test-str_to_rcode.c]: Use tabs for indentation, not spaces.

File: t/test-str_to_rcode.c [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+};
+
+#define num_rcodes (sizeof(rcodes) / sizeof(struct u16str))
+
+static size_t
+test_str_to_rcode(void) {
+	size_t n;
+	size_t failures = 0;
+
+	for(n = 0; n < num_rcodes; n++) {
```
Comment (edmonds) [inline t/test-str_to_rcode.c]: Missing space between `for` and `(`.

File: t/test-str_to_rcode.c [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+}
+
+static int
+check(size_t ret, const char *s)
+{
+        if (ret == 0)
+                fprintf(stderr, NAME ": PASS: %s\n", s);
+        else
+                fprintf(stderr, NAME ": FAIL: %s (%" PRIu64 " failures)\n", s, ret);
+        return (ret);
```
Comment (edmonds) [inline t/test-str_to_rcode.c]: Use tabs instead of spaces in this function.

File: wdns/wdns.h [hunk 4]
```diff
@@ -250,6 +250,9 @@ WDNS_WARN_UNUSED_RESULT
 wdns_res
 wdns_str_to_name(const char *str, wdns_name_t *name);
 
+wdns_res
+wdns_str_to_rcode(const char *str, uint16_t * out);
```
Comment (edmonds) [inline wdns/wdns.h]: Extraneous space between `*` and `out`.

File: wdns/gen_rcode_to_str [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+wdns_res
+wdns_str_to_rcode(const char *str, uint16_t *out) {
+    struct u16str key, *res;
+    key.str = str;
+    res = bsearch(&key, rcodes, num_rcodes, sizeof(struct u16str), cmp_u16str);
+    if (res != NULL) {
+        *out = res->u16;
+        return (wdns_res_success);
+    } else {
```
Comment (edmonds) [inline wdns/gen_rcode_to_str]: DNS doesn't have a generic rcode mnemonic syntax like the generic syntax defined for classes and types. This function should fail (return `wdns_res_failure`) if the string doesn't match a defined rcode, similar to how the inverse function `wdns_rcode_to_str()` fails (returns `NULL`) for unrecognized rcode values.

**Comment by hstern** [comment]
All comments should be addressed now.

**Pull request merged**